R/fit_model.R

```diff
@@ -0,0 +1,42 @@
+#' Fit the RtGam model with {mgcv}
+#'
+#' Use the pre-prepared model dataset and formula. Supply warnings as needed
+#' @noRd
+fit_model <- function(data, formula, backend) {
+  args <- args_constructor(data, formula, backend)
+  call <- call_constructor(backend)
+
+  do.call(
+    call,
+    args
+  )
+}
+
+args_constructor <- function(data, formula, backend) {
+  backend_agnostic_args <- list(
+    formula = formula,
+    data = data,
+    # Negative binomial family with overdispersion param estimated
+    family = "nb"
+  )
+  if (backend == "gam") {
```

Comment:
somewhere there needs to be a match.arg or similar for these gates.

Reply:
https://github.com/CDCgov/cfa-gam-rt/blob/d1f66081211860ab68161df9c7a15e92e92bc33f/R/checkers.R#L113-L118